xdem/spstats.py

```diff
@@ -9,19 +9,92 @@
+    """
+    Estimate an interpolated function for an N-dimensional binning.
+    Relies on scipy.ndimage.map_coordinates: see for more information on input arguments.
+
+    :param df: dataframe of binned values according to explanatory variables
+    :param var_names: list of variable names to use
+    :param statistic: statistic to use
+    :param min_count: filter binned values with a sample count of less than this minimum count
+    :return:
+    """
+
+    # check that the dataframe contains what we need
+    for var in var_names:
+        if var not in df.columns:
+            raise ValueError('Variable "'+var+'" does not exist in the provided dataframe.')
+    statistic_name = statistic if isinstance(statistic,str) else statistic.__name__
+    if statistic_name not in df.columns:
+        raise ValueError('Statistic "' + statistic_name + '" does not exist in the provided dataframe.')
+    if min_count is not None and 'count' not in df.columns:
+        raise ValueError('Statistic "count" is not in the provided dataframe, necessary to use the min_count argument.')
+    if df.empty:
+        raise ValueError('Dataframe is empty.')
+
+    # keep only the binning in the number of dimensions corresponding to the length of the variables
+    df_sub = df[df.nd == len(var_names)]
+    if df_sub.empty:
+        raise ValueError('Dataframe does not contain the number of binned dimensions' + str(len(var_names))+' corresponding to the list of variables.')
+
+    # compute the middle values instead of bin interval
+    for var in var_names:
+        df_sub[var] = pd.IntervalIndex(df_sub[var]).mid.values
+    # keep only rows where the binning data exists for those variables (meaning they were used)
+    df_sub = df_sub[np.logical_and.reduce([np.isfinite(df_sub[var].values) for var in var_names])]
+    if df_sub.empty:
+        raise ValueError('Dataframe does not contain a nd binning with the variables corresponding to the list of variables.')
+
+    # remove statistic values calculated with a sample count under the minimum count
+    if min_count is not None:
+        df_sub.loc[df_sub['count'] < min_count,statistic_name] = np.nan
+
+    # valid values
+    vals = df_sub[statistic_name].values
+    ind_valid = np.isfinite(vals)
+
+    # get a list of middle values for the binning coordinates, to define a nd grid
+    list_bmid = []
+    shape = []
+    for var in var_names:
+        bmid = sorted(np.unique(df_sub[var][ind_valid]))
+        list_bmid.append(bmid)
+        shape.append(len(bmid))
+
+    # griddata first to perform nearest interpolation with NaNs (irregular grid)
+    # valid values
+    vals = vals[ind_valid]
+    # coordinates of valid values
+    points_valid = tuple([df_sub[var].values[ind_valid] for var in var_names])
+    # grid coordinates
+    points_grid = tuple([df_sub[var].values for var in var_names])
+    vals_grid = griddata(points_valid, vals, points_grid, method='nearest')
```

Comment:
Does one always want nearest neighbour interpolation here or is this worth being an argument?

Reply:
Always nearest, see description of the workflow at the top.

Reply:
Ah! I didn't read properly!